doc/releases/changelog-dev.md

```diff
@@ -4,6 +4,72 @@
 
 <h3>New features since last release</h3>
 
+* Added new feature `qml.ops.op_math.Product.py` that was created using Sum.py as a template. This new 
+  feature allows for the creation of a product operation between operators. The product of rotation operatiors is 
+  commutative so the order of the products does not matter. 
+  **Example**
+    The product of qml.PauliX and qml.PauliZ must be equal to -i*qml.PauliY. Which means that the matrix must be equal to 
```

Comment:
Is this from another PR or from yours? I think the indendation here got messed up somehow!